recipes/openssl/1.x.x/conanfile.py

```diff
@@ -635,6 +635,13 @@ def _make(self):
             if self._use_nmake and self._full_version >= "1.1.0":
                 self._replace_runtime_in_file(os.path.join("Configurations", "10-main.conf"))
 
+            if self._use_nmake:
+                # Windows: when cmake generates its cache, it populates some environment variables as well.
+                # If cmake also initiates openssl build, their values (containing spaces and forward slashes)
+                # break nmake (don't know about mingw make). So we fix them
+                for v in ['CC', 'CXX', 'RC']:
+                    if v in os.environ and not '"' in os.environ[v]:
+                        os.environ[v] = '"' + os.environ[v].replace('/', '\\') + '"'
```

Comment:
This modifies the environment permanently.
When using a context, the changes to the environment are undone.

e.g.
```python
from contextlib import contextmanager

@contextmanager
def _build_context(self):
    if self._use_nmake:
        def sanitize_env_var(var):
            return '"{}"'.format(var) if '"' not in var else var
        env = {key: sanitize_env_var(tools.get_env(key)) for key in ("CC", "CXX", "RC") if tools.get_env(key)}
        with tools.environment_append(env):
            yield
    else:
        yield

def build(self):
    with self._build_context():
        self.run("perl ./Configure")
        ...
        self._run_make()
```

Reply:
Yes, it sounds reasonable. But the environment modification should only be done for nmake (it breaks perl configure). I've committed the appropriate changes.